Chromedriver frequently hangs when attempting to start a new session.

Using the standalone-chrome or node-chrome images, I often observe a timeout when attempting to launch a new Chrome session.

Steps to reproduce:

1) launch a standalone chrome instance: `docker run -d --name chrome selenium/standalone-chrome:2.47.1`
2) build the test container: `docker build -t selenium/test:local ./Test`
3) run the repeat test script: `./test-repeat.sh chrome`
4) If necessary, repeat steps 1 and 3 until the problem is observed

Note:

In order to narrow the focus of the test to launching new sessions, you may want to temporarily modify `Test/smoke-test.js` to omit the part of the test that tries navigating to github.com after starting the session. After modifying the script, make sure to rebuild the selenium/test:local image to pick up the changes.

Expected results:

All 50 sessions launch and quit successfully - the test passes.

Actual results:

One of the session launch commands will hang indefinitely.

Docker host: boot2docker v1.7.0 (Tiny Core Linux)

This does look strange:

```
07:44:44.661 INFO - Creating a new session for Capabilities [{rotatable=false, nativeEvents=false, browserName=chrome, takesScreenshot=false, javascriptEnabled=false, version=, platform=ANY, cssSelectorsEnabled=false}]
Starting ChromeDriver 2.20.353124 (035346203162d32c80f1dce587c8154a1efa0c3b) on port 1317
Only local connections are allowed.
...
08:37:00.129 INFO - Command failed to close cleanly. Destroying forcefully (v2). [/opt/selenium/chromedriver-2.20, --port=20521][ {}]
08:37:01.137 ERROR - Unable to kill process with PID 1563
08:37:01.138 WARN - Exception thrown
...
Caused by: org.openqa.selenium.WebDriverException: java.lang.reflect.InvocationTargetException
Driver info: driver.version: unknown
	at org.openqa.selenium.remote.server.DefaultDriverProvider.callConstructor(DefaultDriverProvider.java:113)
...
java.util.concurrent.ExecutionException: org.openqa.selenium.WebDriverException: java.lang.reflect.InvocationTargetException
Build info: version: '2.52.0', ...
...

Driver info: driver.version: ChromeDriver
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:665)
	at org.openqa.selenium.remote.RemoteWebDriver.startSession(RemoteWebDriver.java:249)
	at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:131)
	at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:144)
	at org.openqa.selenium.chrome.ChromeDriver.<init>(ChromeDriver.java:170)
	at org.openqa.selenium.chrome.ChromeDriver.<init>(ChromeDriver.java:138)
	... 14 more
Caused by: org.openqa.selenium.WebDriverException: java.net.SocketTimeoutException: Read timed out
...
Driver info: driver.version: ChromeDriver
	at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:91)
...
Caused by: java.net.SocketTimeoutException: Read timed out
	at java.net.SocketInputStream.socketRead0(Native Method)
...
	at org.openqa.selenium.remote.internal.ApacheHttpClient.execute(ApacheHttpClient.java:90)
	at org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:142)
	at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:82)

```

chromedriver is running there though:

```
docker exec chrome1 ps aux|grep chromedriver
seluser   3931  0.0  0.1 391412  9292 ?        Sl   07:44   0:00 /opt/selenium/chromedriver-2.20 --port=1317
seluser   4689  0.0  0.0 391412  7432 ?        Sl   08:23   0:00 /opt/selenium/chromedriver-2.20 --port=30084
```

But port isn't opened:

```
 docker exec chrome1 lsof -i :1317
```
shows nothing

--- a/Makefile
+++ b/Makefile
@@ -227,10 +227,10 @@ release: tag_major_minor
 	docker push $(NAME)/standalone-firefox-debug:$(MAJOR_MINOR_PATCH)
 
 test:
-	./test.sh
-	./sa-test.sh
-	./test.sh debug
-	./sa-test.sh debug
+	VERSION=$(VERSION) ./test.sh
+	VERSION=$(VERSION) ./sa-test.sh
+	VERSION=$(VERSION) ./test.sh debug
+	VERSION=$(VERSION) ./sa-test.sh debug
 
 .PHONY: \
 	all \

--- a/NodeBase/Dockerfile.txt
+++ b/NodeBase/Dockerfile.txt
@@ -10,7 +10,6 @@ RUN apt-get update -qqy \
   && apt-get -qqy install \
     locales \
     xvfb \
-    dbus \
   && rm -rf /var/lib/apt/lists/* /var/cache/apt/*
 
 #==============================

--- a/NodeChrome/Dockerfile.txt
+++ b/NodeChrome/Dockerfile.txt
@@ -57,15 +57,14 @@ COPY generate_config /opt/selenium/generate_config
 RUN chmod +x /opt/selenium/generate_config
 
 #=================================
-# Chrome Launch Script Modication
+# Chrome Launch Script Modification
 #=================================
 COPY chrome_launcher.sh /opt/google/chrome/google-chrome
 RUN chmod +x /opt/google/chrome/google-chrome
 
 RUN chown -R seluser:seluser /opt/selenium
 
-# Following line fixes
-# https://github.com/SeleniumHQ/docker-selenium/issues/87
-RUN echo "DBUS_SESSION_BUS_ADDRESS=/dev/null" >> /etc/environment
+# Following line fixes https://github.com/SeleniumHQ/docker-selenium/issues/87
+ENV DBUS_SESSION_BUS_ADDRESS=/dev/null
 
 USER seluser

--- a/NodeFirefox/Dockerfile.txt
+++ b/NodeFirefox/Dockerfile.txt
@@ -53,8 +53,7 @@ COPY generate_config /opt/selenium/generate_config
 RUN chmod +x /opt/selenium/generate_config \
   && chown -R seluser:seluser /opt/selenium
 
-# Following line fixes
-# https://github.com/SeleniumHQ/docker-selenium/issues/87
-RUN echo "DBUS_SESSION_BUS_ADDRESS=/dev/null" >> /etc/environment
+# Following line fixes https://github.com/SeleniumHQ/docker-selenium/issues/87
+ENV DBUS_SESSION_BUS_ADDRESS=/dev/null
 
 USER seluser

--- a/sa-test.sh
+++ b/sa-test.sh
@@ -1,6 +1,7 @@
 #!/usr/bin/env bash
 
 DEBUG=''
+VERSION=${VERSION:-3.4.0-bismuth}
 
 if [ -n "$1" ] && [ $1 == 'debug' ]; then
   DEBUG='-debug'
@@ -13,7 +14,7 @@ function test_standalone {
   BROWSER=$1
   echo Starting Selenium standalone-$BROWSER$DEBUG container
 
-  SA=$(docker run -d selenium/standalone-$BROWSER$DEBUG:3.4.0-bismuth)
+  SA=$(docker run -d selenium/standalone-$BROWSER$DEBUG:${VERSION})
   SA_NAME=$(docker inspect -f '{{ .Name  }}' $SA | sed s:/::)
   TEST_CMD="node smoke-$BROWSER.js"
 

--- a/test.sh
+++ b/test.sh
@@ -24,7 +24,7 @@ echo Building test container image
 docker build -t selenium/test:local ./Test
 
 echo 'Starting Selenium Hub Container...'
-HUB=$(docker run -d selenium/hub:3.4.0-bismuth)
+HUB=$(docker run -d selenium/hub:${VERSION})
 HUB_NAME=$(docker inspect -f '{{ .Name  }}' $HUB | sed s:/::)
 echo 'Waiting for Hub to come online...'
 docker logs -f $HUB &

